samples/gocdk-docstore/main.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// gocdk-docstore demonstrates the use of the Go CDK docstore package in a
+// simple command-line application.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"time"
+
+	"github.com/google/subcommands"
+	"gocloud.dev/internal/docstore"
+
+	// Import the docstore driver packages we want to be able to open.
+	_ "gocloud.dev/internal/docstore/dynamodocstore"
+	_ "gocloud.dev/internal/docstore/firedocstore"
+	_ "gocloud.dev/internal/docstore/memdocstore"
+	_ "gocloud.dev/internal/docstore/mongodocstore"
+)
+
+const helpSuffix = `
+
+  See https://gocloud.dev/concepts/urls/ for more background on
+  Go CDK URLs, and sub-packages under gocloud.dev/internal/docstore
+  (https://godoc.org/gocloud.dev/internal/docstore#pkg-subdirectories)
+  for details on the docstore.Collection URL format.
+`
+
+func main() {
+	subcommands.Register(subcommands.HelpCommand(), "")
+	subcommands.Register(&listCmd{}, "")
+	subcommands.Register(&putCmd{}, "")
+	subcommands.Register(&deleteCmd{}, "")
+	log.SetFlags(0)
+	log.SetPrefix("gocdk-docstore: ")
+	flag.Parse()
+	os.Exit(int(subcommands.Execute(context.Background())))
+}
+
+type Message struct {
+	Date             string
```

Comment:
Why are these both strings and not `time.Time`s?

Reply:
I use `Timestamp` as the key, firestore only takes string for the name field. Filed #2331.

Reply:
Oh, right.

Reply:
On second thought, I think you should use something else as the primary key. Timestamp isn't good because it's not guaranteed to be unique. It also has the two problems I mentioned—it is redundant with date, and people will wonder why it's not a time.Time (I don't know if we'll fix that for Firestore, because I don't think timestamps work as primary keys).

So how about generating a unique ID using the `github.com/google/uuid` package? 
```
type Message {
    Key string // unique random key
    Date string // date message was added, as a string for easy searching by date
   Content string
   ...
}
```

Reply:
Done.